lib/stdlib/src/binary.erl

```diff
@@ -365,31 +365,39 @@ get_opts_replace(_,_) ->
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% Hex encoding functions
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
--define(HEX(X), (hex(X)):16).
--compile({inline,[hex/1]}).
+
 -spec encode_hex(Bin) -> Bin2 when
       Bin :: binary(),
       Bin2 :: <<_:_*16>>.
-encode_hex(Data) when byte_size(Data) rem 8 =:= 0 ->
-    << <<?HEX(A),?HEX(B),?HEX(C),?HEX(D),?HEX(E),?HEX(F),?HEX(G),?HEX(H)>> || <<A,B,C,D,E,F,G,H>> <= Data >>;
-encode_hex(Data) when byte_size(Data) rem 7 =:= 0 ->
-    << <<?HEX(A),?HEX(B),?HEX(C),?HEX(D),?HEX(E),?HEX(F),?HEX(G)>> || <<A,B,C,D,E,F,G>> <= Data >>;
-encode_hex(Data) when byte_size(Data) rem 6 =:= 0 ->
-    << <<?HEX(A),?HEX(B),?HEX(C),?HEX(D),?HEX(E),?HEX(F)>> || <<A,B,C,D,E,F>> <= Data >>;
-encode_hex(Data) when byte_size(Data) rem 5 =:= 0 ->
-    << <<?HEX(A),?HEX(B),?HEX(C),?HEX(D),?HEX(E)>> || <<A,B,C,D,E>> <= Data >>;
-encode_hex(Data) when byte_size(Data) rem 4 =:= 0 ->
-    << <<?HEX(A),?HEX(B),?HEX(C),?HEX(D)>> || <<A,B,C,D>> <= Data >>;
-encode_hex(Data) when byte_size(Data) rem 3 =:= 0 ->
-    << <<?HEX(A),?HEX(B),?HEX(C)>> || <<A,B,C>> <= Data >>;
-encode_hex(Data) when byte_size(Data) rem 2 =:= 0 ->
-    << <<?HEX(A),?HEX(B)>> || <<A,B>> <= Data >>;
-encode_hex(Data) when is_binary(Data) ->
-    << <<?HEX(N)>> || <<N>> <= Data >>;
 encode_hex(Bin) ->
+    encode_hex(Bin, upper).
+
+-define(HEX(X, Case), (hex(X, Case)):16).
+-compile({inline,[hex/2]}).
+-spec encode_hex(Bin, Case) -> Bin2 when
+      Bin :: binary(),
+      Case :: lower | upper,
+      Bin2 :: <<_:_*16>>.
```

Comment:
Would `lowercase` / `uppercase` be better namings than `upper` / `lower`? Not sure what the precedents are.

Reply:
Yes, `lowercase`/`uppercase` is better.